lib/srv/desktop/windows_server.go

```diff
@@ -921,6 +927,18 @@ func (s *WindowsService) makeTDPSendHandler(ctx context.Context, emitter events.
 				// it on the TDP connection
 				s.onClipboardReceive(ctx, emitter, id, sessionID, desktopAddr, int32(len(clip)))
 			}
+		case byte(tdp.TypeSharedDirectoryAcknowledge):
```

Comment:
General observation, not related to your specific changes. This "send handler" runs in the critical path of processing a message like a bitmap. The longer it takes to run, the long it will be before we can emit the next PNG. We should measure this at some point and see if handling the auditing out of band in a background goroutine makes a difference.

Reply:
Good call, turned this comment into an issue: https://github.com/gravitational/teleport/issues/17487